### DISCUSSION TITLE
How do I use a http backend binarycaching

### DISCUSSION THREAD

**Discussion by xiaozhuai** [microsoft/vcpkg#28353] [Q&A]
I've notice that vcpkg has http binarycaching backend supported, and it's not documented yet.
I tried this feature and didn't get it work.

I have a webdav server `https://nas.xxx.com` and an account `username:password`.
So I came up this.
```
vcpkg install hiredis:x64-osx --binarysource=http,https://username:password@nas.xxx.com/data/vcpkg/{triplet}/{name}/{version}/{sha},readwrite
```

It fails with http code 403

```
Replace secretes produced the following error: 'Error: curl failed to put file to https://username:password@nas.xxx.com/data/vcpkg/x64-osx/hiredis/1.0.2/b21d90c65501d331ee87d136e20e5d4fbe253383d1b07bb562ffa60abbc6bfbb with exit code '0' and http code '403'
```

**Comment by autoantwort** [comment]
Does it work if you try it manually on the command line?

**Comment by Osyotr** [comment]
You probably need to pass `Authorization: Basic <auth>` after `readwrite`, where `<auth>` is the result of `echo -n "username:password" | base64`

**Reply by xiaozhuai** [reply]
Tried with no luck, maybe there are some problem with my webdav server. I'll check later.
